saga_2/src/saga_core/saga_cmd/man/saga_cmd.1
.\" Automatically generated by Pod::Man 2.1801 (Pod::Simple 3.07)
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  \*(C+ will
.\" give a nicer C++.  Capital omega is used to do unbreakable dashes and
.\" therefore won't be available.  \*(C` and \*(C' expand to `' in nroff,
.\" nothing in troff, for use with C<>.
.tr \(*W-
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
'br\}
.\"
.\" Escape single quotes in literal strings from groff's Unicode transform.
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\"
.\" If the F register is turned on, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.SS), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.ie \nF \{\
.    de IX
.    tm Index:\\$1\t\\n%\t"\\$2"
..
.    nr % 0
.    rr F
.\}
.el \{\
.    de IX
..
.\}
.\"
.\" Accent mark definitions (@(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2).
.\" Fear.  Run.  Save yourself.  No user-serviceable parts.
.    \" fudge factors for nroff and troff
.if n \{\
.    ds #H 0
.    ds #V .8m
.    ds #F .3m
.    ds #[ \f1
.    ds #] \fP
.\}
.if t \{\
.    ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.    ds #V .6m
.    ds #F 0
.    ds #[ \&
.    ds #] \&
.\}
.    \" simple accents for nroff and troff
.if n \{\
.    ds ' \&
.    ds ` \&
.    ds ^ \&
.    ds , \&
.    ds ~ ~
.    ds /
.\}
.if t \{\
.    ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.    ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.    ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.    ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.    ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.    ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.\}
.    \" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.    \" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.    \" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.    ds : e
.    ds 8 ss
.    ds o a
.    ds d- d\h'-1'\(ga
.    ds D- D\h'-1'\(hy
.    ds th \o'bp'
.    ds Th \o'LP'
.    ds ae ae
.    ds Ae AE
.\}
.rm #[ #] #H #V #F C
.\" ========================================================================
.\"
.IX Title "SAGA_CMD 1"
.TH SAGA_CMD 1 "2010-03-08" "2.0.4" " "
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.if n .ad l
.nh
.SH "NAME"
\&\fBsaga_cmd\fR \- command-line interface to the System for Automated Geoscientific Analyses
.SH "SYNOPSIS"
.IX Header "SYNOPSIS"
[\fB\s-1SAGA_MLB\s0\fR=\fIdirname\fR] \fBsaga_cmd\fR \fI\s-1LIBRARY\s0\fR [\fI\s-1MODULE\s0\fR] [\fB\-silent\fR] <module specific options...>
.PP
\&\fBsaga_cmd\fR [\fB\-h, \-\-help\fR]
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
Command line interface to the System for Automated Geoscientific Analyses (\s-1SAGA\s0). When run without arguments, the module libraries in the current module library are listed. This can be specified by setting the environment variable \fI\s-1SAGA_MLB\s0\fR. If this is not specified, the current working directory is used.
.PP
\&\s-1SAGA\s0 \s-1GIS\s0 (System for Automated Geoscientific Analysis) is a free and open source geographic information system used for editing and analysing spatial data. It includes a large number of modules for the analysis of vector, table, grid and image data.
.SH "OPTIONS"
.IX Header "OPTIONS"
.IP "[\fI\s-1SAGA_MLB\s0\fR=dirname]" 8
.IX Item "[SAGA_MLB=dirname]"
Module libraries are expected to be located in the directory that is specified by the environment variable '\s-1SAGA_MLB\s0'. If this is not found the current working directory will be searched for instead.
.Sp
To use the default modules, the environment variable can be set to \fI/usr/lib/saga\fR
.IP "\fI\s-1LIBRARY\s0\fR [\fI\s-1MODULE\s0\fR]" 8
.IX Item "LIBRARY [MODULE]"
Specify the library (ie ta_morphometry) and the module (number or name). If no module is specified the different modules of a library are listed.
.IP "\fB\-h, \-\-help\fR" 8
.IX Item "-h, --help"
Display help and exit
.SH "EXAMPLE"
.IX Header "EXAMPLE"
export SAGA_MLB=/usr/lib/saga
.PP
\&\fBsaga_cmd\fR ta_morphometry \*(L"Local Morphometry\*(R" \-ELEVATION dem.sgrd \-SLOPE slope.sgrd \-ASPECT aspect.sgrd \-METHOD 1
.PP
Multiple input files can be specified by seperating them by a semicolon (;).
.PP
\&\fBsaga_cmd\fR libgrid_tools Merging \-GRIDS test1.sgrd\e;test2.sgrd \-GRID_TARGET merged.sgrd
.SH "BUGS"
.IX Header "BUGS"
On the website of \fIhttp://www.saga\-gis.org\fR, a list of known bugs can be found and new bug reports can be submitted.
.SH "AUTHORS"
.IX Header "AUTHORS"
\&\fB\s-1SAGA\s0\fR was written by Olaf Conrad with help from a large number of contributers. This manual page was written by Johan Van de Wauw.
Both are released under the \s-1GNU\s0 General Public License v2.
.PP
More info  \fIhttp://www.saga\-gis.org\fR


saga_2/NEWS
_______________________________________________________________________________

         SAGA 2.0 - System for Automated Geoscientific Analyses
_______________________________________________________________________________

NEWS
_______________________________________________________________________________

* May, 15th 2007

 - release of SAGA 2.0



saga_2/src/saga_core/saga_cmd/saga_cmd.cpp

///////////////////////////////////////////////////////////
//                                                       //
//                         SAGA                          //
//                                                       //
//      System for Automated Geoscientific Analyses      //
//                                                       //
//                Command Line Interface                 //
//                                                       //
//                   Program: SAGA_CMD                   //
//                                                       //
//-------------------------------------------------------//
//                                                       //
//                     SAGA_CMD.cpp                      //
//                                                       //
//          Copyright (C) 2005 by Olaf Conrad            //
//                                                       //
//-------------------------------------------------------//
//                                                       //
// This file is part of 'SAGA - System for Automated     //
// Geoscientific Analyses'. SAGA is free software; you   //
// can redistribute it and/or modify it under the terms  //
// of the GNU General Public License as published by the //
// Free Software Foundation; version 2 of the License.   //
//                                                       //
// SAGA is distributed in the hope that it will be       //
// useful, but WITHOUT ANY WARRANTY; without even the    //
// implied warranty of MERCHANTABILITY or FITNESS FOR A  //
// PARTICULAR PURPOSE. See the GNU General Public        //
// License for more details.                             //
//                                                       //
// You should have received a copy of the GNU General    //
// Public License along with this program; if not,       //
// write to the Free Software Foundation, Inc.,          //
// 59 Temple Place - Suite 330, Boston, MA 02111-1307,   //
// USA.                                                  //
//                                                       //
//-------------------------------------------------------//
//                                                       //
//    contact:    Olaf Conrad                            //
//                Institute of Geography                 //
//                University of Goettingen               //
//                Goldschmidtstr. 5                      //
//                37077 Goettingen                       //
//                Germany                                //
//                                                       //
//    e-mail:     oconrad@saga-gis.org                   //
//                                                       //
///////////////////////////////////////////////////////////

//---------------------------------------------------------


///////////////////////////////////////////////////////////
//                                                       //
//                                                       //
//                                                       //
///////////////////////////////////////////////////////////

//---------------------------------------------------------
#include <wx/app.h>
#include <wx/utils.h>
#include <wx/dir.h>
#include <wx/filename.h>

#include <saga_api/saga_api.h>

#include "callback.h"

#include "module_library.h"


///////////////////////////////////////////////////////////
//														 //
//														 //
//														 //
///////////////////////////////////////////////////////////

//---------------------------------------------------------
#define SAGA_CMD_VERSION	SG_T("2.0.5")

#define SYS_ENV_PATH		SG_T("PATH")

#define FLAG_SILENT		SG_T("s")
#define FLAG_INTERACT		SG_T("i")
#define FLAG_PROJ			SG_T("p")
#define FLAG_LANGUAGE		SG_T("l")


///////////////////////////////////////////////////////////
//														 //
//														 //
//														 //
///////////////////////////////////////////////////////////

//---------------------------------------------------------
bool		Execute			(const SG_Char *MLB_Path, const SG_Char *FileName, const SG_Char *ModuleName, int argc, char *argv[]);

void		Error_Library	(const SG_Char *MLB_Path);
void		Error_Module	(const SG_Char *MLB_Path, const SG_Char *FileName);

void		Print_Logo		(void);
void		Print_Execution	(const SG_Char *MLB_Path, const SG_Char *FileName, const SG_Char *ModuleName, const SG_Char *Author);
void		Print_Help		(void);

void		Create_Example	(void);


///////////////////////////////////////////////////////////
//														 //
//														 //
//														 //
///////////////////////////////////////////////////////////

//---------------------------------------------------------
int		main	(int argc, char *argv[])
{
//---------------------------------------------------------
#if !defined(_DEBUG) && defined(_SAGA_VC)
#define _MODULE_EXCEPTION
_try 
{
#endif
//---------------------------------------------------------

	if( !wxInitialize() )
	{
		Print_Error(SG_T("initialisation failed"));

		return( 1 );
	}

	//-----------------------------------------------------
	wxString	Flags, CMD_Path, MLB_Path, ENV_Path;

	if( argc > 1 )
	{
		CSG_String	s(argv[1]);

		if( !s.CmpNoCase(SG_T("-h")) || !s.CmpNoCase(SG_T("--help")) )
		{
			Print_Help();

			return( 0 );
		}

		if( !s.CmpNoCase(SG_T("-b")) || !s.CmpNoCase(SG_T("--batch")) )
		{
			Create_Example();

			return( 0 );
		}

		s	= s.BeforeFirst(SG_T('='));

		if( !s.CmpNoCase(SG_T("-f")) || !s.CmpNoCase(SG_T("--flags")) )
		{
			Flags	= CSG_String(argv[1]).AfterFirst(SG_T('=')).c_str();

			argc--;
			argv++;
		}
	}

	//-----------------------------------------------------
	CMD_Path	= SG_File_Get_Path(SG_UI_Get_Application_Path()).c_str();

	if( !wxGetEnv(SG_T("SAGA_MLB"), &MLB_Path) || MLB_Path.Length() == 0 )
	{
		MLB_Path	= SG_File_Make_Path(CMD_Path, SG_T("modules"))  .c_str();
    #if defined( _SAGA_LINUX)
        MLB_Path = wxT(MODULE_LIBRARY_PATH);
    #endif
	}


	if( wxGetEnv(SYS_ENV_PATH, &ENV_Path) && ENV_Path.Length() > 0 )
	{
		wxSetEnv(SYS_ENV_PATH, wxString::Format(wxT(";%s;%s"), MLB_Path.c_str(), SG_File_Make_Path(CMD_Path, SG_T("dll")).c_str()));
	}
	else
	{
		wxSetEnv(SYS_ENV_PATH, wxString::Format(wxT( "%s;%s"), MLB_Path.c_str(), SG_File_Make_Path(CMD_Path, SG_T("dll")).c_str()));
	}

	//-----------------------------------------------------
	SG_Set_UI_Callback(Get_Callback());

	Set_Silent		(Flags.Find(FLAG_SILENT  ) >= 0 ? true : false);
	Set_Interactive	(Flags.Find(FLAG_INTERACT) >= 0 ? true : false);

	Print_Logo();

	if( Flags.Find(FLAG_LANGUAGE) > 0 )
	{
		SG_Get_Translator() .Create(SG_File_Make_Path(CMD_Path, SG_T("saga"), SG_T("lng")), false);
	}

	if( Flags.Find(FLAG_PROJ) > 0 )
	{
		SG_Get_Projections().Create(SG_File_Make_Path(CMD_Path, SG_T("saga"), SG_T("srs")));
	}

	//-----------------------------------------------------
	bool	bResult	= false;

	switch( argc )
	{
	case 1: 
		Error_Library		(MLB_Path);
		break;

	case 2:
		Error_Module		(MLB_Path, SG_STR_MBTOSG(argv[1]));
		break;

	default:
		bResult	= Execute	(MLB_Path, SG_STR_MBTOSG(argv[1]), SG_STR_MBTOSG(argv[2]), argc - 2, argv + 2);
		break;
	}

	//-----------------------------------------------------
	if( ENV_Path.Length() > 0 )
	{
		wxSetEnv(SYS_ENV_PATH, ENV_Path);
	}
	else
	{
		wxUnsetEnv(SYS_ENV_PATH);
	}

	wxUninitialize();

//---------------------------------------------------------
#ifdef _DEBUG
	Set_Interactive(true);
	Get_Pause();
#endif

#ifdef _MODULE_EXCEPTION
}
_except(1)
{
	Print_Error(LNG("access violation"));
}
#endif
//---------------------------------------------------------

	return( bResult ? 0 : 1 );
}


///////////////////////////////////////////////////////////
//														 //
//														 //
//														 //
///////////////////////////////////////////////////////////

//---------------------------------------------------------
bool		Execute(const SG_Char *MLB_Path, const SG_Char *FileName, const SG_Char *ModuleName, int argc, char *argv[])
{
	CModule_Library	Library;

	if( !Library.Create(FileName, MLB_Path) )
	{
		Error_Library(MLB_Path);
	}
	else if( !Library.Select(ModuleName) )
	{
		Print_Error(LNG("module not found"), ModuleName);

		Error_Module(MLB_Path, FileName);
	}
	else if( Library.Get_Selected()->is_Interactive() )
	{
		Print_Error(LNG("cannot execute interactive module"), ModuleName);

		Error_Module(MLB_Path, FileName);
	}
	else
	{
		Print_Execution(MLB_Path, FileName, Library.Get_Selected()->Get_Name(), Library.Get_Selected()->Get_Author());

		Set_Library(&Library);

		if( Library.Execute(argc, argv) )
		{
			Set_Library(NULL);

			return( true );
		}

		Set_Library(NULL);
	}

	return( false );
}


///////////////////////////////////////////////////////////
//                                                       //
//                                                       //
//                                                       //
///////////////////////////////////////////////////////////

//---------------------------------------------------------
void		Error_Library	(const SG_Char *MLB_Path)
{
	int				nLibraries;
	wxDir			Dir;
	wxString		FileName;
	CModule_Library Library;

	if( !Dir.Open(MLB_Path) )
	{
		Print_Error(LNG("invalid module libraries path"), MLB_Path);
	}
	else if(	!Dir.GetFirst(&FileName, wxT("*.dll"), wxDIR_FILES|wxDIR_HIDDEN)
			&&	!Dir.GetFirst(&FileName, wxT("*.so" ), wxDIR_FILES|wxDIR_HIDDEN) )
	{
		Print_Error(LNG("no valid module library found in path"), MLB_Path);
	}
	else
	{
		Print_Error(LNG("module library"));

		if( !Get_Silent() )
		{
			SG_PRINTF(SG_T("\n%s:\n"), LNG("available module libraries"));

			nLibraries	= 0;

			do
			{
				if( Library.Create(FileName, Dir.GetName()) )
				{
					SG_PRINTF(SG_T("- %s\n"), FileName.c_str());
					nLibraries++;
				}
			}
			while( Dir.GetNext(&FileName) );

			SG_PRINTF(SG_T("\n%d %s\n"), nLibraries, LNG("SAGA Module Libraries"));
		}
	}

	if( !Get_Silent() )
	{
		SG_PRINTF(SG_T("\n"));
		SG_PRINTF(LNG("type -h or --help for further information"));
		SG_PRINTF(SG_T("\n"));
	}
}

//---------------------------------------------------------
void		Error_Module	(const SG_Char *MLB_Path, const SG_Char *FileName)
{
	CModule_Library	Library;

	if( !Library.Create(FileName, MLB_Path) )
	{
		Library.Destroy();

		Print_Error(LNG("module library not found"), FileName);

		Error_Library(MLB_Path);

		return;
	}

	Print_Error(LNG("module"));

	if( !Get_Silent() )
	{
		SG_PRINTF(SG_T("\n%s:\n"), LNG("available modules"));

		for(int i=0; i<Library.Get_Count(); i++)
		{
			if( Library.Get_Module(i)->is_Interactive() )
			{
				SG_PRINTF(SG_T("[%d]\t- [%s] %s\n"), i, LNG("interactive"), Library.Get_Module(i)->Get_Name());
			}
			else
			{
				SG_PRINTF(SG_T(" %d\t- %s\n"), i, Library.Get_Module(i)->Get_Name());
			}
		}

		SG_PRINTF(SG_T("\n"));
		SG_PRINTF(LNG("type -h or --help for further information"));
		SG_PRINTF(SG_T("\n"));
	}
}


///////////////////////////////////////////////////////////
//                                                       //
//                                                       //
//                                                       //
///////////////////////////////////////////////////////////

//---------------------------------------------------------
void		Print_Logo		(void)
{
	if( Get_Silent() )
		return;

	SG_PRINTF(SG_T("_____________________________________________\n"));
	SG_PRINTF(SG_T("  #####   ##   #####    ##\n"));
	SG_PRINTF(SG_T(" ###     ###  ##       ###\n"));
	SG_PRINTF(SG_T("  ###   # ## ##  #### # ##\n"));
	SG_PRINTF(SG_T("   ### ##### ##    # #####\n"));
	SG_PRINTF(SG_T("##### #   ##  ##### #   ##\n"));
	SG_PRINTF(SG_T("_____________________________________________\n"));
	SG_PRINTF(SG_T("\n"));
}

//---------------------------------------------------------
void		Print_Execution	(const SG_Char *MLB_Path, const SG_Char *FileName, const SG_Char *ModuleName, const SG_Char *Author)
{
	if( Get_Silent() )
		return;

	SG_PRINTF(SG_T("%s:\t%s\n"), LNG("library path"), MLB_Path);
	SG_PRINTF(SG_T("%s:\t%s\n"), LNG("library name"), FileName);
	SG_PRINTF(SG_T("%s:\t%s\n"), LNG("module name "), ModuleName);
	SG_PRINTF(SG_T("%s:\t%s\n"), LNG("author      "), Author);
	SG_PRINTF(SG_T("_____________________________________________\n"));
	SG_PRINTF(SG_T("go...\n"));
}

//---------------------------------------------------------
void		Print_Help		(void)
{
	Print_Logo();

	SG_PRINTF(
		SG_T("SAGA API ") SAGA_API_VERSION SG_T("\n")
		SG_T("SAGA CMD ") SAGA_CMD_VERSION SG_T("\n")
		SG_T("under GNU General Public License (GPL)\n")
		SG_T("O.Conrad (C) 2005-10\n")
		SG_T("\n")
		SG_T("Usage:\n")
		SG_T("\n")
		SG_T("saga_cmd [-h, --help]\n")
		SG_T("saga_cmd [-b, --batch]\n")
		SG_T("saga_cmd [-f, --flags][=silp] <LIBRARY> <MODULE> <module specific options...>\n")
		SG_T("\n")
		SG_T("[-h], [--help ]: help on usage\n")
		SG_T("[-b], [--batch]: create a batch file example\n")
		SG_T("[-f], [--flags]: various flags for general usage\n")
		SG_T("  s: silent mode\n")
		SG_T("  i: allow user interaction\n")
		SG_T("  l: load translation dictionary\n")
		SG_T("  p: load projections dictionary\n")
		SG_T("<LIBRARY>\t")	SG_T(": file name of the library\n")
		SG_T("<MODULE>\t")	SG_T(": either name or index of the module\n")
		SG_T("\n")
		SG_T("example:\n")
		SG_T("  saga_cmd -f=sp ta_morphometry \n")
		SG_T("           \"Local Morphometry\"\n")
		SG_T("           -ELEVATION c:\\dem.sgrd\n")
		SG_T("           -SLOPE     d:\\slope.sgrd\n")
		SG_T("           -ASPECT    d:\\aspect.sgrd\n")
		SG_T("           -METHOD    1\n")
		SG_T("\n")
		SG_T("_____________________________________________\n")
		SG_T("Module libraries are expected to be in the SAGA installation\n")
		SG_T("directory or its \'modules\' subdirectory. If this is not found\n")
		SG_T("the current working directory will be searched for instead.\n")
		SG_T("Alternatively you can add the environment variable \'SAGA_MLB\'\n")
		SG_T("and let it point to the desired directory\n")
		SG_T("\n")
		SG_T("SAGA CMD is particularly useful for the automated\n")
		SG_T("execution of a series of analysis steps, because it\n")
		SG_T("allows you to execute modules using batch files.\n")
		SG_T("Calling saga_cmd with the option \'-b\' or \'--batch\'\n")
		SG_T("creates a batch file example. You probably have to edit\n")
		SG_T("the path definitions to make the batch file run on your\n")
		SG_T("computer.\n")
	);
}


///////////////////////////////////////////////////////////
//														 //
//														 //
//														 //
///////////////////////////////////////////////////////////

//---------------------------------------------------------
void		Create_Example	(void)
{
	CSG_String	FileName(CSG_String::Format(SG_T("%s\\saga_cmd_example.bat"), wxGetCwd().c_str()));
	CSG_File	Stream;

	SG_PRINTF(SG_T("\n%s...\n"), LNG("creating batch file example"));

	if( !Stream.Open(FileName, SG_FILE_W, false) )
	{
		SG_PRINTF(SG_T("%s\n"), LNG("failed"));

		return;
	}

	Stream.Printf(
		SG_T("@echo off\n")
		SG_T("\n")
		SG_T("set SAGA=.\n")
		SG_T("set SAGA_MLB=%%SAGA%%\\modules\n")
		SG_T("PATH=PATH;%%SAGA%%;%%SAGA_MLB%%\n")
		SG_T("\n")
		SG_T("if exist .\\srtm.asc goto :SRTM\n")
		SG_T("echo _____________________________________________\n")
		SG_T("echo import SRTM-DEM from ESRI ASCII grid\n")
		SG_T("saga_cmd.exe recreations_fractals 5 -GRID .\\dem.sgrd -NX 400 -NY 400 -H 0.75\n")
		SG_T("\n")
		SG_T("goto :GO\n")
		SG_T("\n")
		SG_T(":SRTM\n")
		SG_T("echo _____________________________________________\n")
		SG_T("echo import SRTM-DEM from ESRI ASCII grid\n")
		SG_T("saga_cmd.exe io_grid              1 -FILE .\\srtm.asc -GRID .\\srtm.sgrd\n")
		SG_T("\n")
		SG_T("echo _____________________________________________\n")
		SG_T("echo projection from geodetic to UTM zone 32\n")
		SG_T("\n")
		SG_T("saga_cmd.exe pj_proj4             1 -SOURCE .\\srtm.sgrd -OUT_GRID .\\dem.sgrd -PROJ_TYPE 109 -utm_zone 32 -TARGET_TYPE 1 -GET_AUTOFIT_GRIDSIZE 90.0\n")
		SG_T("\n")
		SG_T(":GO\n")
		SG_T("echo _____________________________________________\n")
		SG_T("echo create contour lines from DEM\n")
		SG_T("\n")
		SG_T("saga_cmd.exe shapes_grid          5 -INPUT .\\dem.sgrd -CONTOUR .\\contour.shp -ZSTEP 25.0\n")
		SG_T("\n")
		SG_T("echo _____________________________________________\n")
		SG_T("echo do some terrain analysis\n")
		SG_T("\n")
		SG_T("saga_cmd.exe ta_preprocessor      1 -DEM .\\dem.sgrd -DEM_PREPROC .\\dem.sgrd\n")
		SG_T("saga_cmd.exe ta_lighting          0 -ELEVATION .\\dem.sgrd -SHADE .\\shade.sgrd -METHOD 0 -AZIMUTH -45 -DECLINATION 45\n")
		SG_T("saga_cmd.exe ta_morphometry       0 -ELEVATION .\\dem.sgrd -SLOPE .\\slope.sgrd -ASPECT .\\aspect.sgrd -HCURV .\\hcurv.sgrd -VCURV .\\vcurv.sgrd\n")
		SG_T("saga_cmd.exe ta_hydrology         0 -ELEVATION .\\dem.sgrd -CAREA .\\carea.sgrd\n")
		SG_T("saga_cmd.exe ta_hydrology        14 -SLOPE .\\slope.sgrd -AREA .\\carea.sgrd -WETNESS .\\wetness.sgrd -STREAMPOW .\\streampow.sgrd -LSFACTOR .\\lsfactor.sgrd\n")
		SG_T("\n")
		SG_T("echo _____________________________________________\n")
		SG_T("echo perform cluster analysis\n")
		SG_T("\n")
		SG_T("saga_cmd.exe grid_discretisation  1 -INPUT .\\dem.sgrd;.\\slope.sgrd;.\\hcurv.sgrd;.\\vcurv.sgrd -RESULT .\\cluster.sgrd -STATISTICS .\\cluster.txt -NORMALISE -NCLUSTER 10\n")
		SG_T("\n")
		SG_T("pause\n")
	);

	SG_PRINTF(SG_T("%s\n"), LNG("okay"));
}


///////////////////////////////////////////////////////////
//														 //
//														 //
//														 //
///////////////////////////////////////////////////////////

//---------------------------------------------------------
